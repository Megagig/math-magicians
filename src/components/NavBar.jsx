import { NavLink } from 'react-router-dom';

const Navbar = () => {
  const links = [
    { path: '/', text: 'Home' },
    { path: 'quotes', text: 'Quotes' },
    { path: 'calculator', text: 'Calculator' },
  ];

  const footerStyles = {
    position: 'fixed',
    bottom: '0px',
    right: '10px',
  };

  return (
    <>
      <nav className="navbar">
        <h1>Math Magicians</h1>
        <ul>
          {links.map((link) => (
            <li key={link.path}>
              <NavLink to={link.path}>{link.text}</NavLink>
            </li>
          ))}
        </ul>
      </nav>
      <footer style={footerStyles}>
        Crafted with code and creativity by
        {' '}
        <b>Obi Anthony</b>
        .
      </footer>
    </>
  );
};

export default Navbar;
