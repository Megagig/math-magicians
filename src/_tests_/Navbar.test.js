import React from 'react';
import { render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import 'regenerator-runtime/runtime';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';

import NavBar from '../components/NavBar';

describe('NavBar component', () => {
  test('renders NavBar with correct button labels', () => {
    render(
      <BrowserRouter>
        <NavBar />
      </BrowserRouter>,
    );

    const buttonLabels = ['Home', 'Quotes', 'Calculator'];

    buttonLabels.forEach((label) => {
      const buttonElement = screen.getByText(label);
      expect(buttonElement).toBeInTheDocument();
    });
  });
});
describe('NavBar snapshot', () => {
  test('matches the snapshot', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <NavBar />
        </BrowserRouter>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
