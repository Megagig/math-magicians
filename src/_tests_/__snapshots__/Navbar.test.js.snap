// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar snapshot matches the snapshot 1`] = `
Array [
  <nav
    className="navbar"
  >
    <h1>
      Math Magicians
    </h1>
    <ul>
      <li>
        <a
          aria-current="page"
          className="active"
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
      <li>
        <a
          className=""
          href="/quotes"
          onClick={[Function]}
        >
          Quotes
        </a>
      </li>
      <li>
        <a
          className=""
          href="/calculator"
          onClick={[Function]}
        >
          Calculator
        </a>
      </li>
    </ul>
  </nav>,
  <footer
    style={
      Object {
        "bottom": "0px",
        "position": "fixed",
        "right": "10px",
      }
    }
  >
    All Right Reserve @Megagig 2023
  </footer>,
]
`;
